text/0000-clippy-uno.md

```diff
@@ -0,0 +1,263 @@
+
+Lints can only belong to one lint group at a time, and the lint group defines the lint level. There is a bunch of overlap between
+the style and complexity groups -- a lot of style issues are also complexity issues and vice versa. We separate these groups
+so that people can opt in to the complexity lints without having to opt in to Clippy's style.
+
+## Compiler uplift
+
+The compiler has historically had a "no new lints" policy, partly with the desire that lints would
+incubate outside of Clippy. This feels like a good time to look into uplifting these lints.
+
+This RFC does not _yet_ propose lints to be uplifted, but the intention is that the RFC
+discussion will bring up lints that the community feels _should_ be uplifted and we can list them here.
+
+Such an uplift may change the lint level; correctness lints are Deny
+by default in Clippy but would probably switch to Warn if uplifted since the compiler is more
+conservative here (Using clippy is in itself an opt-in to a "please annoy me more" mode).
+
+
+We'd also like to establish a rough policy for future lints here:  Some correctness lints should probably belong in the compiler,
+whereas style/complexity/etc lints should probably belong in Clippy. Lints may be incubated in Clippy, of course.
+
+I don't think the compler will want _all_ correctness lints here, however if the lint is about a common enough situation
+where it being _not_ a bug is an exceedingly rare case (i.e. very low false positive frequency) it should probably belong in the
+compiler.
+
+ [cat]: #lint-categorization
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+
+## Lint categorization
+
+This categorization can be browsed [online].
+
+ [online]: http://rust-lang-nursery.github.io/rust-clippy/current/
+
+Please leave comments on thoughts about these lints -- if their categorization is correct, if they should exist at all, and if we should be uplifting them to the compiler.
+
+
+### correctness (Deny)
+* [naive_bytecount](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#naive_bytecount) (use of naive `<slice>.filter(|&x| x == y).count()` to count byte values)
+* [map_entry](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#map_entry) (use of `contains_key` followed by `insert` on a `HashMap` or `BTreeMap`)
+* [boxed_local](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#boxed_local) (using `Box<T>` where unnecessary)
+* [large_enum_variant](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#large_enum_variant) (large size difference between variants on an enum)
+* [manual_memcpy](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#manual_memcpy) (manually copying items between slices)
+* [unused_collect](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#unused_collect) (`collect()`ing an iterator without using the result; this is usually better written as a for loop)
```

Comment:
It's a good lint; For the suggestion part, I think using `.for_each(|elt| { .. })` is also idiomatic.

Reply:
I think specifics like this should be raised as issues on the repo, not here :smile:

We're dealing with the broad idea of the lints here, not the specific ones.

(This RFC can become huge in scope otherwise)

Reply:
Yeah that makes sense. 😆